graphblas/matrix.py

```diff
@@ -740,11 +740,23 @@ def apply(self, op, right=None, *, left=None):
         Apply UnaryOp to each element of the calling Matrix
         A BinaryOp can also be applied if a scalar is passed in as `left` or `right`,
             effectively converting a BinaryOp into a UnaryOp
+        An IndexUnaryOp can also be applied with the thunk passed in as `right`
         """
         method_name = "apply"
         extra_message = (
-            "apply only accepts UnaryOp with no scalars or BinaryOp with `left` or `right` scalar."
+            "apply only accepts UnaryOp with no scalars or BinaryOp with `left` or `right` scalar "
+            "or IndexUnaryOp with `right` thunk."
         )
+        if isinstance(op, str):
+            op = op_from_string(op)
+        opclass = find_opclass(op)[1]
+        if opclass in ("IndexUnaryOp", "SelectOp"):
```

Comment:
Small nit: I now prefer sets instead of tuples, so like ` in {"IndexUnaryOp", "SelectOp"}`